src/Components/WebAssembly/Server/src/TargetPickerUi.cs

```diff
@@ -37,6 +40,182 @@ public TargetPickerUi([StringSyntax(StringSyntaxAttribute.Uri)] string debugProx
+                    }
+                }
+                if (!string.IsNullOrEmpty(consoleActorId))
+                {
+                    if (res?["input"]?.Value<string>()?.StartsWith("AboutDebugging.actions.addNetworkLocation(", StringComparison.InvariantCultureIgnoreCase) == true)
+                    {
+                        await EvaluateOnBrowser(toStream, consoleActorId, $"if (AboutDebugging.store.getState().runtimes.networkRuntimes.find(element => element.id == \"{_debugProxyUrl}\").runtimeDetails !== null) {{ AboutDebugging.actions.selectPage(\"runtime\", \"{_debugProxyUrl}\"); if (AboutDebugging.store.getState().runtimes.selectedRuntimeId == \"{_debugProxyUrl}\") AboutDebugging.actions.inspectDebugTarget(\"tab\", {tabToRedirect})}};", token);
+                    }
+                    if (res?["input"]?.Value<string>()?.StartsWith("if (AboutDebugging.store.getState()", StringComparison.InvariantCultureIgnoreCase) == true)
+                    {
+                        await EvaluateOnBrowser(toStream, consoleActorId, $"if (AboutDebugging.store.getState().runtimes.networkRuntimes.find(element => element.id == \"{_debugProxyUrl}\").runtimeDetails !== null) {{ AboutDebugging.actions.selectPage(\"runtime\", \"{_debugProxyUrl}\"); if (AboutDebugging.store.getState().runtimes.selectedRuntimeId == \"{_debugProxyUrl}\") AboutDebugging.actions.inspectDebugTarget(\"tab\", {tabToRedirect})}};", token);
+                    }
+                }
+                else
+                {
+                    if (!string.IsNullOrEmpty(res?["target"]?["consoleActor"]?.Value<string>()))
+                    {
+                        consoleActorId = res["target"]?["consoleActor"]?.Value<string>();
+                        await EvaluateOnBrowser(toStream, consoleActorId, $"AboutDebugging.actions.addNetworkLocation(\"{_debugProxyUrl}\"); AboutDebugging.actions.connectRuntime(\"{_debugProxyUrl}\");", token);
+                    }
+                    else if (!string.IsNullOrEmpty(res?["actor"]?.Value<string>()))
+                    {
+                        await SendMessageToBrowser(toStream, JObject.FromObject(new { type = "watchTargets", targetType = "frame", to = res?["actor"]?.Value<string>() }), token);
+                        await SendMessageToBrowser(toStream, JObject.FromObject(new { type = "watchResources", resourceTypes = new JArray("console-message"), to = res?["actor"]?.Value<string>() }), token);
+                    }
+                    else if (!string.IsNullOrEmpty(toCmd))
+                    {
+                        await SendMessageToBrowser(toStream, JObject.FromObject(new { type = "getWatcher", isServerTargetSwitchingEnabled = true, to = toCmd }), token);
+                    }
```

Comment:
Is this definitely the right way to factor the code? Since debugging-WebAssembly-on-Firefox isn't Blazor-specific (it should ideally be possible even if you're not using Blazor at all), wouldn't it be more applicable to have this code in the runtime library that contains the debug proxy itself?

Apologies if I'm misunderstanding the layering here!

Reply:
Also alt-shift-d for chrome is not Blazor specific, so this code should also be moved.
I added here to be near to the code that does the same for chrome.

Reply:
We don't have a way to get all the tabs available for debugging using a simple request as we do for chrome. We need to use the firefox remote debugging protocol to do it.

Reply:
OK, up to you to decide the best place for it. If - longer term - we can factor out the non-Blazor-specific parts into a runtime package that sounds ideal. Would it be worth adding an issue to track that task?

Reply:
https://github.com/dotnet/runtime/issues/81028